common/message_algos/src/grouping/retirement_policy.rs

```diff
@@ -0,0 +1,21 @@
+use crate::{grouping::MessageGroup, Message, Timestamp};
+
+/// Decision on retiring message groups ("batches").
+pub trait RetirementPolicy: Send {
```

Comment:
As per the batching proposal, `batch timeout` is the one and only retirement policy. So, not sure if this trait is really needed. But, I like that abstraction, though.